### ISSUE TITLE
Empty catch block and no-empty, // Do nothing doesn't help

### ISSUE THREAD

**Issue by SimenB** [eslint/eslint#3315]
Wow, a really bad title... Suggestions? :laughing: 

I'm having an issue with the `no-empty`-rule. From the [docs](http://eslint.org/docs/rules/no-empty.html#rule-details), putting in `// Do nothing` should make it not error, but I'm still getting the error...

```js
export default {
  removeSelection: () => {
    let selection;

    if ('getSelection' in window) {
      selection = window.getSelection();
    } else {
      selection = ('selection' in document) ? document.selection : null;
    }

    try {
      if ('removeAllRanges' in selection) {
        selection.removeAllRanges();
      } else if ('empty' in selection) {
        selection.empty();
      }
    } catch (ex) { // <- Throws error
      // Do nothing
    }
  }
};
```

Using `eslint@1.0.0`
eslint output: 
```
$ node_modules/.bin/eslint common-assets/src/assets/scripts/meep.js 

common-assets/src/assets/scripts/meep.js
  17:18  error  Empty block statement  no-empty

✖ 1 problem (1 error, 0 warnings)
```

config: 
```json
{
  "root": true,
  "extends": "eslint-config-airbnb",
  "rules": {
    "comma-dangle": [
      2,
      "never"
    ],
    "func-names": 0,
    "space-before-function-paren": [
      2,
      "always"
    ],
    "no-control-regex": 2,
    "no-dupe-args": 2,
    "no-empty-character-class": 2,
    "no-extra-boolean-cast": 2,
    "no-negated-in-lhs": 2,
    "no-regex-spaces": 2,
    "valid-jsdoc": [
      1,
      {
        "requireReturn": false
      }
    ],
    "valid-typeof": 2,
    "dot-location": [
      2,
      "property"
    ],
    "no-alert": 2,
    "no-empty-label": 2,
    "no-labels": 2,
    "no-useless-call": 2,
    "no-delete-var": 2,
    "no-label-var": 2,
    "no-undef-init": 2,
    "no-unused-vars": 2,
    "no-use-before-define": 2,
    "indent": [2, 2, {"SwitchCase": 1}],
    "array-bracket-spacing": 2,
    "newline-after-var": 2,
    "no-lonely-if": 2,
    "no-unneeded-ternary": 2,
    "object-curly-spacing": [
      2,
      "always"
    ],
    "operator-assignment": [
      2,
      "always"
    ],
    "space-unary-ops": 1,
  }
}
```

**Comment by eslintbot** [comment]
Thanks for the issue! We get a lot of issues, so this message is automatically posted to each one to help you check that you've included all of the information we need to help you.

Reporting a bug? Please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
1. The source code that caused the problem
1. The configuration you're using (for the rule or your entire config file)
1. The actual ESLint output complete with line numbers

Requesting a new rule? Please be sure to include:

1. The use case for the rule - what is it trying to prevent or flag?
1. Whether the rule is trying to prevent an error or is purely stylistic
1. Why you believe this rule is generic enough to be included

Requesting a feature? Please be sure to include:

1. The problem you want to solve (don't mention the solution)
1. Your take on the correct solution to problem

Including this information in your issue helps us to triage it and get you a response as quickly as possible.

Thanks!

**Comment by ilyavolodin** [comment]
Your .eslintrc doesn't show it, but are you using `babel-eslint` by any chance?

**Comment by SimenB** [comment]
`airbnb` config uses babel, yeah

**Comment by ilyavolodin** [comment]
I think babel-eslint is stripping comments. At least it was at some point. Could you test the same setup with Espree?

**Comment by SimenB** [comment]
You are correct. See babel/babel-eslint#141
Seems like it's a `wontfix` following the links... :cry:
